lib/ansible/modules/cloud/vmware/vmware_host.py

```diff
@@ -199,17 +282,17 @@ def main():
     argument_spec = vmware_argument_spec()
     argument_spec.update(
         datacenter_name=dict(type='str', required=True),
-        cluster_name=dict(type='str'),
+        cluster_name=dict(type='str', required=True),
         esxi_hostname=dict(type='str', required=True),
-        esxi_username=dict(type='str', required=True),
-        esxi_password=dict(type='str', required=True, no_log=True),
-        state=dict(type='str', default='present', choices=['absent', 'present'])
-    )
+        esxi_username=dict(type='str', required=False),
```

Comment:
esxi_username, esxi_password are required options, so maybe you can use  ```required_if```

Reply:
I didn't know `required_if`, thanks for the hint. I did the modification.